### ISSUE TITLE
Unable to create missing rules in VS 2019

### ISSUE THREAD

**Issue by mikehachen** [bchavez/Bogus#230]
I bought a bogus premium license and creating missing rules works in VS 2017. However when I try to create the rules in VS 2019 (16.1.3) I get the following error:
```
System.InvalidOperationException : Sequence contains no elements
   at System.Linq.Enumerable.First[TSource](IEnumerable 1 source)
   at Bogus.Tools.Analyzer.MissingRules.ProbeForGoodVariable(ObjectCreationExpressionSyntax oce,SemanticModel model)
   at async Bogus.Tools.Analyzer.MissingRules.CodeFixAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetPreviewOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.GetPreviewResultAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedActionWithNestedFlavors.<>c__DisplayClass11_0.<GetPreviewAsync>b__0(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](<Unknown Parameters>)
```

Any suggestions?

Thanks, Mike

**Comment by bchavez** [comment]
Hi Mike,

Thanks for raising the issue. What version of the Analyzer are you using?

**Comment by mikehachen** [comment]
Hi 

Sorry for not providing this important information... :( I use the latest available version, 27.0.1.6

**Comment by bchavez** [comment]
Thanks! I'll see what I can do to reproduce the issue on my end.

However, I have a feeling this might be something specific inside your source file and `Faker<T>` usage.

It would help a lot if you could provide a sample of your `Faker<T>` code either by posting an anonymized version here, or you can email me directly bchavez@bitarmory.com. Any extra info will help.

It's a bit late here in the US-Los Angeles side, but I'll have a quick look before I go to bed and see if I spot anything. But for sure tomorrow, I'll set a block of time to figure this one out; hopefully, have a fix for you soon.

**Comment by mikehachen** [comment]
Thank you for your help!

I'll try to create a sample project. Below you'll find the code where this error happens:

`       [Fact, AutoRollback]
        public void SaveNewAddress()
        {
            //Arrange
            var processor = CreateAddressMaintenanceProcessor();
            var address = new Faker<Address>("de");
            //Act
            var result = processor.SaveAddress(address);

            //Assert
            Assert.NotNull(result);
            Assert.Equal(address.Address01, result.Address01);
            Assert.Equal(address.Address02, result.Address02);
            Assert.Equal(address.Address03, result.Address03);
            Assert.Equal(address.AddressReference, result.AddressReference);
            Assert.Equal(address.AddressStatus, result.AddressStatus);
            Assert.Equal(address.AddressType, result.AddressType);
            Assert.Equal(address.City, result.City);
            Assert.Equal(address.ContactType, result.ContactType);
            Assert.Equal(address.CountryCode, result.CountryCode);
            Assert.Equal(DateTime.Today, result.CreatedAt.Date);
            Assert.Equal(address.CreditLimit, result.CreditLimit);
            Assert.Equal(address.CurrencyCode, result.CurrencyCode);
            Assert.Equal(address.DefaultPurchaseConditionCode, result.DefaultPurchaseConditionCode);
            Assert.Equal(address.DefaultSalesConditionCode, result.DefaultSalesConditionCode);
            Assert.Equal(address.DefaultTextId, result.DefaultTextId);
            Assert.Equal(address.DeliveryAddressId, result.DeliveryAddressId);
            Assert.Equal(address.DeliveryTypeId, result.DeliveryTypeId);
            Assert.Equal(address.DunningBlock, result.DunningBlock);
            Assert.Equal(address.DunningDate.HasValue, result.DunningDate.HasValue);
            Assert.Equal(address.DunningLevel, result.DunningLevel);
            Assert.Equal(address.Email, result.Email);
            Assert.Equal($"{address.Firstname} ", result.Firstname, ignoreLineEndingDifferences: true);
            Assert.Equal(address.InvoiceAddressId, result.InvoiceAddressId);
            Assert.Equal(address.IsAddressChanged, result.IsAddressChanged);
            Assert.Equal(address.IsPartDeliveryAllowed, result.IsPartDeliveryAllowed);
            Assert.Equal(address.LanguageCode, result.LanguageCode);
            Assert.Equal(address.Lastname, result.Lastname);
            Assert.Equal(address.LetterSalutation, result.LetterSalutation);
            Assert.Equal(DateTime.Today, result.ModifiedAt.Date);
            Assert.Equal("Web API", result.ModifiedBy);
            Assert.Equal(address.NoOfInvoiceCopy, result.NoOfInvoiceCopy);
            Assert.Equal(address.Notes, result.Notes);
            Assert.Equal(address.ParentId, result.ParentId);
            Assert.Equal(address.Postalcode, result.Postalcode);
            Assert.Equal(address.PriceListId, result.PriceListId);
            Assert.Equal(address.ProjectCostCenterId, result.ProjectCostCenterId);
            Assert.Equal(address.PurchasePrices, result.PurchasePrices);
            Assert.Equal(address.RayonId, result.RayonId);
            Assert.Equal(address.Region, result.Region);
            Assert.Equal(address.RemoteId, result.RemoteId);
            Assert.Equal(address.Salutation, result.Salutation);
            Assert.Equal(address.ServiceAddressId, result.ServiceAddressId);
            Assert.Equal(address.StorageLocationId, result.StorageLocationId);
            Assert.Equal(address.Telefax, result.Telefax);
            Assert.Equal(address.Telephone01, result.Telephone01);
            Assert.Equal(address.Telephone02, result.Telephone02);
            Assert.Equal(address.Telephone03, result.Telephone03);
            Assert.Equal(address.VatTypeId, result.VatTypeId);
            Assert.Equal(address.Website, result.Website);
            Assert.Equal(address.MasterId, result.MasterId);
            Assert.Equal(address.ExternalId, result.ExternalId);
            Assert.Equal(address.ExternalString01, result.ExternalString01);
            Assert.Equal(address.ExternalString02, result.ExternalString02);
            Assert.Equal(address.ExternalString03, result.ExternalString03);
            Assert.Equal(address.ExternalString04, result.ExternalString04);
            Assert.Equal(address.ExternalString05, result.ExternalString05);
            Assert.Equal(address.ExternalString06, result.ExternalString06);
            Assert.Equal(address.ExternalString07, result.ExternalString07);
            Assert.Equal(address.ExternalString08, result.ExternalString08);
            Assert.Equal(address.ExternalString09, result.ExternalString09);
            Assert.Equal(address.ExternalString10, result.ExternalString10);
        }
`

**Comment by bchavez** [comment]
Okay, I was able to reproduce the error and trigger the exception.

Basically, if you only have a `.Generate()` method after `Faker<T>` and no rules, similar to the following code below:

```csharp
var faker2 = new Faker<Address>("de")
                   .Generate();
```
It will throw:
```csharp
System.InvalidOperationException : Sequence contains no elements
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at Bogus.Tools.Analyzer.MissingRules.ProbeForGoodVariable(ObjectCreationExpressionSyntax oce,SemanticModel model)
   at async Bogus.Tools.Analyzer.MissingRules.CodeFixAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetPreviewOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.GetPreviewResultAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedActionWithNestedFlavors.<>c__DisplayClass11_0.<GetPreviewAsync>b__0(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](<Unknown Parameters>)
```

If you remove the `.Generate()` method call and just have `var faker2 = new Faker<Address>("de");`, then the analyzer should work as expected.

So, having the `.Generate()` method immediately after `Faker<T>` without rules will trip up the analyzer. You might need to restart VS to get the analyzer working again after the first exception.

I should have a fix for you soon. I'm not sure I can make a release tonight (almost 1AM here) because I'm half alseep. I don't want to wake up to a disaster :fire: :fire_engine: if I make a mistake right now. :smile: 

Rest assured, I will have a fix for you in the next 24 hours.

**Comment by mikehachen** [comment]
Hey Brian

Ah, such a simple solution! Great, thank you for looking into this even at 1AM!

If you have a fix by next Monday I'm more than happy! There's a workaround so no need for a possible disaster... ;)

Have a nice weekend!
Michael

**Comment by bchavez** [comment]
Hi Michael,

As promised, I just released `v27.0.1.7` that contains the fix for trailing methods that cause the analyzer to throw an exception. You can pick it up on NuGet [here](https://www.nuget.org/packages/Bogus.Tools.Analyzer/27.0.1.7).

![capture-1](https://user-images.githubusercontent.com/478118/60376110-832ccd00-99c1-11e9-9128-37a6b542dde7.gif)

Also, I tested the analyzer in **VS 2017 (15.9.13)** and **VS 2019 (16.1.3)** and it seems to be all working with trailing (non-returning `Faker<T>`) methods.

Please let me know if the update works for you. I'll leave the issue open until you can confirm it's fixed on your end. Also, don't forget to check my email I sent earlier about the `Address(Local, Random)` properties that were being generated.

Again, thanks for your support. My apologies for the delay!

:briefcase: :necktie: ***["Taking care of business every day... Taking care of business every way..."](https://www.youtube.com/watch?v=mmwic9kFx2c)***

**Comment by mikehachen** [comment]
Hi Brian

Thank you very much for your support and quick fix! It works perfectly, so the issue can be closed :) 

Wish you a nice weekend and thanks again
Michael
